### PR TITLE
Fixing lint issues in the signature package

### PR DESCRIPTION
**Pull request by susperius**
#### Summary
Fixing pylint and pytype issues in the signature package

#### Release Note

None

#### Documentation

None

### CODE REVIEW

File: model_signing/signature/encoding.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 PAYLOAD_TYPE = "application/vnd.in-toto+json"
 
+# The proto contains a Statement message.
+# pylint: disable=no-member
+
```
Comment (mihaimaruseac) [inline model_signing/signature/encoding.py:21]: Can we disable this locally and re-enable as soon as we no longer need it disabled?

File: model_signing/signature/key.py [hunk 2]
```diff
@@ -95,12 +95,13 @@ def from_path(cls, key_path: str):
 
     def verify(self, bundle: bundle_pb.Bundle) -> None:
         statement = json_format.Parse(
-            bundle.dsse_envelope.payload, statement_pb.Statement())
+            bundle.dsse_envelope.payload,
+            statement_pb.Statement())  # pylint: disable=no-member
```
Comment (mihaimaruseac) [inline model_signing/signature/key.py:99]:
```suggestion
            statement_pb.Statement()  # pylint: disable=no-member
        )
```

File: model_signing/signature/key.py [hunk 3]
```diff
@@ -95,12 +95,13 @@ def from_path(cls, key_path: str):
...
         pae = encoding.pae(statement)
         try:
             self._public_key.verify(
                 bundle.dsse_envelope.signatures[0].sig,
                 pae, ec.ECDSA(SHA256()))
         except Exception as e:
             raise VerificationError(
-                'signature verification failed ' + str(e))
+                'signature verification failed ' + str(e)) from e
```
Comment (mihaimaruseac) [inline model_signing/signature/key.py:107]:
```suggestion
                'signature verification failed ' + str(e)
            ) from e
```

File: model_signing/signature/pki.py [hunk 4]
```diff
@@ -152,7 +154,7 @@ def verify(self, bundle: bundle_pb.Bundle) -> None:
             store_ctx.verify_certificate()
         except ssl_crypto.X509StoreContextError as err:
             raise VerificationError(
-                f'signing certificate verification failed: {err}')
+                f'signing certificate verification failed: {err}') from err
```
Comment (mihaimaruseac) [inline model_signing/signature/pki.py:157]:
```suggestion
                f'signing certificate verification failed: {err}'
            ) from err
```

File: model_signing/signature/sigstore.py [hunk 5]
```diff
@@ -35,7 +35,10 @@ class SigstoreSigner(Signer):
 
     CLIENT_ID = "sigstore"
 
-    def __init__(self, disable_ambient: bool = True, id_provider: str = None):
+    def __init__(
+            self,
+            disable_ambient: bool = True,
+            id_provider: str | None = None):
```
Comment (mihaimaruseac) [inline model_signing/signature/sigstore.py:41]:
```suggestion
            id_provider: str | None = None
        ):
```

File: model_signing/signature/sigstore.py [hunk 6]
```diff
@@ -99,4 +102,4 @@ def verify(self, bundle: bundle_pb.Bundle) -> None:
             sig_bundle = sig_models.Bundle(bundle)
             _ = self._verifier.verify_dsse(sig_bundle, self._policy)
         except Exception as e:
-            raise VerificationError(str(e))
+            raise VerificationError(str(e)) from e
```
Comment (mihaimaruseac) [inline model_signing/signature/sigstore.py:105]:
```suggestion
            raise VerificationError(str(e)
        ) from e
```

File: model_signing/signature/key.py (hunk 2 above)
Comment (mihaimaruseac) [reply model_signing/signature/key.py:99]: This is to try and be consistent with the python style, but given that the linter didn't pick it up and we don't yet have an autoformatter, it's mostly a nit.

**Review by mihaimaruseac — COMMENTED** [review]
Thank you! This LGTM.

Added a few nits to simplify work later when we add more CI support.

File: model_signing/signature/encoding.py [hunk 7]
```diff
@@ -17,7 +17,8 @@
 PAYLOAD_TYPE = "application/vnd.in-toto+json"
 
 
-def pae(statement: statement_pb.Statement) -> bytes:
+def pae(statement: statement_pb.Statement  # pylint: disable=no-member
+        ) -> bytes:
```
Comment (mihaimaruseac) [inline model_signing/signature/encoding.py:21]:
```suggestion
def pae(
    statement: statement_pb.Statement  # pylint: disable=no-member
) -> bytes:
```

File: model_signing/signature/sigstore.py [hunk 8]
```diff
@@ -99,4 +103,5 @@ def verify(self, bundle: bundle_pb.Bundle) -> None:
             sig_bundle = sig_models.Bundle(bundle)
             _ = self._verifier.verify_dsse(sig_bundle, self._policy)
         except Exception as e:
-            raise VerificationError(str(e))
+            raise VerificationError(str(e)
+        ) from e
```
Comment (mihaimaruseac) [inline model_signing/signature/sigstore.py:107]:
```suggestion
            raise VerificationError(str(e)) from e
```

**Review by mihaimaruseac — COMMENTED** [review]
Thank you!

**Comment by mihaimaruseac** [comment]
Oh, it seems trying to speed up merging by committing code suggestions actually makes it so that my review doesn't count for merging. Let's wait for @spencerschrock or @laurentsimon for one more review for merging this.

Sorry for the delay.

**Review by spencerschrock — APPROVED** [review]

**Pull request merged**